internal/services/devcenter/dev_center_gallery_resource_gen.go

```diff
@@ -0,0 +1,189 @@
+package devcenter
+
+// NOTE: this file is generated - manual changes will be overwritten.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See NOTICE.txt in the project root for license information.
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devcenter/2023-04-01/galleries"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+var _ sdk.Resource = DevCenterGalleryResource{}
+
+type DevCenterGalleryResource struct{}
+
+func (r DevCenterGalleryResource) ModelObject() interface{} {
+	return &DevCenterGalleryResourceSchema{}
+}
+
+type DevCenterGalleryResourceSchema struct {
+	DevCenterId       string `tfschema:"dev_center_id"`
+	GalleryResourceId string `tfschema:"gallery_resource_id"`
```

Comment:
Like we did for the generated resource let's rename this to `shared_gallery_id`
```suggestion
	GalleryResourceId string `tfschema:"shared_gallery_id"`
```